src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/illegalidentifiername/InputIllegalIdentifierNameUnnamedVariables.java

```diff
@@ -0,0 +1,46 @@
+/*
+IllegalIdentifierName
+format = (default)(?i)^(?!(record|yield|var|permits|sealed)$).+$
+tokens = (default)CLASS_DEF, INTERFACE_DEF, ENUM_DEF, ANNOTATION_DEF, ANNOTATION_FIELD_DEF, \
+         PARAMETER_DEF, VARIABLE_DEF, METHOD_DEF, ENUM_CONSTANT_DEF, PATTERN_VARIABLE_DEF, \
+         RECORD_DEF, RECORD_COMPONENT_DEF, LAMBDA
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.naming.illegalidentifiername;
+
+public class InputIllegalIdentifierNameUnnamedVariables {
+    void m(Object o) {
+        int _ = sideEffect();
+        for (Integer _ : new int[0]) {
```

Comment:
Please add other examples like `__` (double underscore) and `_BAD` and `BAD_`. I just want to verify that they won't be picked up.

Reply:
done